packages/editor/src/bindings/post-data.js

```diff
@@ -79,7 +114,16 @@ export default {
 		}
 		return newValues;
 	},
-	setValues( { dispatch, context, bindings } ) {
+	setValues( { dispatch, context, bindings, clientId, select } ) {
+		const { getBlockName } = select( blockEditorStore );
+
+		const blockName = getBlockName?.( clientId );
+
+		// Navigaton block types are read-only.
+		// See https://github.com/WordPress/gutenberg/pull/72165.
+		if ( NAVIGATION_BLOCK_TYPES.includes( blockName ) ) {
+			return false;
```

Comment:
AFAICS, `setValues` doesn't have a return type, so we can simply do

```suggestion
			return;
```